clusterloader2/testing/density/config.yaml

```diff
@@ -69,12 +72,39 @@ steps:
     Params:
       action: start
       labelSelector: kind = latencyPod
+{{range $iteration := (RangeInt 1 $latencyPodsIterations)}}
```

Comment:
I hate it - it starts looking like coding in yaml.

Why do we need iterations?
Why can't we simply created all latency pods in a single iteration?
I don't see any reason why we would need iterations.

Reply:
> I hate it - it starts looking like coding in yaml.

I agree. However I'm not sure what's the alternative.

> Why do we need iterations?

I would like to implement (more or less) the same logic as in original test - this way it's easier to compare the results.

> Why can't we simply created all latency pods in a single iteration?

That's tricky. Assuming that we already have a lot of saturation pods, adding 1 pod is not a problem. Adding 5 can exceed the quota. (Needs to be verified.)
It also depends on what we want to achieve and what we want to measure. Creating 5 times 20 pods and deleting them is different than creating and deleting 100 pods.

Reply:
So I can tell you what the motivation for that was.
In general, if there were two few pods, a single pods may affect 99th percentile. So we want to schedule bit enough number of those to avoid such situation.

TBH, I'm not convinced that what exactly was done is the best way. And while in general I agree replicating the test existing is better, we should do it carefully.

In this particular case, I really think that we don't want to do iteration - this precedent is too dangerous.
If you remember, I already told you in the past that I'm against such configs, and I would prefer hardcoding stuff. But I gave up.
But this is waaay to much for me.

re quota issues - it's about maxPodsPerNode, not quota.
And for all tests we run, we would be fine even with 5 pods.

Let's just change it to schedule max(500, #nodes) latency pods and not introduce iterations.

In fact, I still believe this test is too complicated due to all those fancy parameters that noone will never change...

Reply:
Iteration removed.
The only problem is that max(500, #nodes) works really bad if someone uses it with 5-nodes cluster.

Reply:
This test has other problems for small clusters...

Let's add a comment that this test is supposed to run on clusters larger than X (50?, 100?) and that's fine.

Reply:
Comment added.

> This test has other problems for small clusters...

The results was meaningless in terms of scalability - the sample was to small, right?
However, it was good as an example - everyone could run it on any cluster.